support/cas-server-support-consent-webflow/src/main/java/org/apereo/cas/web/consent/CasConsentReviewController.java

```diff
@@ -0,0 +1,144 @@
+package org.apereo.cas.web.consent;
+
+import org.apereo.cas.consent.ConsentDecision;
+import org.apereo.cas.consent.ConsentEngine;
+import org.apereo.cas.consent.ConsentRepository;
+import org.apereo.cas.web.support.WebUtils;
+import org.apereo.inspektr.common.spi.PrincipalResolver;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.context.J2EContext;
+import org.pac4j.core.engine.CallbackLogic;
+import org.pac4j.core.http.J2ENopHttpActionAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+/**
+ * This is {@link CasConsentReviewController}.
+ *
+ * @author Arnold Bergner
+ * @since 5.2.0
+ */
+@Controller("casConsentReviewController")
+@RequestMapping("/consent")
```

Comment:
`/consent` is a little too generic; it would be best if this was made a bit more explicit by using something like `/contextReview`, etc.